### ISSUE TITLE
No documentation for how to make the dist folder. 

### ISSUE THREAD

**Issue by tomandersen** [wet-boew/wet-boew#6660]
We want to build a WET website. I have used bootstrap, scss, ruby, php, C++, etc. 

1) download the base theme
2) now what? the docs say to edit the html files (What? the base dist download is the output), but there are none, and I am pretty sure that I should be editing the hbs files under site and then running some code that will build the site in the 'dist' folder. Which we have done with ruby/Sinatra, etc on other sites.
3) I run scripts/setup - lots happens, a few warnings. 
Still no dist folder, so I run
4) grunt
Which I can see by reading the docs for grunt and looking at the grunt coffee script file should make the dist folder. It fails on my OS X machine with 

Snow-Pro:theme-base-4.0.12 tandersen$ grunt
Loading "Gruntfile.coffee" tasks...ERROR
>> Error: Unable to read "lib/jquery/bower.json" file (Error code: ENOENT).
Warning: Task "assemble" not found. Use --force to continue.

Aborted due to warnings.

But since there is no documentation on this aspect of development - I'm not even sure you are supposed to run grunt (or grunt server?) for local development. 

I would like to know if I'm on the right track, but it looks like to me that the issue is in the documentation.

**Comment by nschonni** [comment]
Did you check the getting started guide? http://wet-boew.github.io/wet-boew/docs/start-en.html

**Comment by tomandersen** [comment]
I was following the guide. 

Do I type 'grunt' at the command line to build the distribution files? 

Do I type 'grunt server' to get a server up that watches for changes in the .hbs files?

Where is the documentation for how to install and develop a web site with the WET toolkit. The guide page laughably says to edit the html files, when the whole point of a build system is to have header, footer templates, etc to build a static distribution folder, a design process we have used in the past.

**Comment by LaurentGoderre** [comment]
UNless you want to change the template or the Javascript, you don't need to build anything. You can download the already built themes here (the production files column):

http://wet-boew.github.io/wet-boew/docs/versions/dwnld-en.html#v40stable

**Comment by tomandersen** [comment]
The themes download page has theme 'dist' folders, which are quite useless, as they are html files. We want to edit .hbs headers, footers, scss, javascript, etc to build a bootstrap site that is accessibility standard compliant. 

The base theme download of source require one to follow (I think as there is no documentation) the steps to get to the point where 'grunt server' works and serves files. 

Basically, I need a blank website with a header, footer, etc hbs and a few template pages.

I am trying "I want to customize WET and design a new WET theme" and removing the wet-boew dependency as that is a lot of baggage that we will not be using (or rather we will only use bits of it as required).

**Comment by tomandersen** [comment]
Anyway a stock install of the "I want to customize WET and design a new WET theme" fails with 

         Running "sass:all" (sass) task
         Warning: /Users/tandersen/code/waterproject/lib/wet-boew/src/plugins/geomap/ie8:5: error: file to import not found or unreadable: "sprites/sprites_geomap"
         Use --force to continue.

         Aborted due to warnings.

         Execution Time (2015-04-01 20:00:43 UTC)

**Comment by LaurentGoderre** [comment]
We do have a Yeoman generator that can assist you in creating a new theme and it is the recommended way.

https://github.com/wet-boew/generator-wet-boew-theme

Let me know if you need help with the generator

**Comment by at88mph** [comment]
@tomandersen, have you made progress with this?

I just wanted to share some of my experience as I've been creating a new theme with 4.0.12 (latest release as of this writing).  I have a Debian 8.0 (testing) system with grunt-cli v0.1.13.

To touch on your build issue, I had the same problem.  I ran the ```script/setup```, and it failed during the post install as you described.  Then I ran ```grunt init``` since it was in the log output of the failure and seemed to be where it was coming from.  It failed on jscs (JavaScript Code Style checker), so I ran ```grunt init --force```, and that passed.  After that, it was smooth sailing for builds just by running ```grunt``` on its own.

As for theme creation in general, there seem to be two ways to create a new theme:

1. Download the base one, as you've done, and kind of pick things out and feel your way around until you have something mostly functional.  I did this with the 3.1.x version, and it was OK, but not intuitive.

1. Use the theme generator and take it from bare metal.  This one seems like a great idea at first, but unless you have good SCSS/SASS knowledge, it's challenging, but a good challenge.  I really recommend learning a lot about SCSS anyway (http://youtu.be/onDepQs4-uk).

Unfortunately, there is a big gap between these two methods; one may be too basic, the other may be too advanced.  What I've done is use the bare metal version, and used the base theme for reference whenever I change something, just to make sure it's kosher(ish).

Good luck.  It's a steep learning curve.
Dustin

**Comment by LaurentGoderre** [comment]
Any update on this?

**Comment by at88mph** [comment]
Hi Laurent,

I’m afraid not.  Sorry.

Dustin


Dustin N. Jenkins 
Tel/Tél: 250.363.3101 | facsimile/télécopieur: (250) 363-0045 | Dustin.Jenkins@nrc-cnrc.gc.ca 
National Research Council Canada | 5071 West Saanich Rd, Victoria BC. V9E 2E7 
Conseil national de recherches Canada | 5071, ch. West Saanich, Victoria (C.-B) V9E 2E7 
Government of Canada | Gouvernement du Canada

**Comment by hsrudnicki** [comment]
Can this issue be closed or otherwise?

**Comment by at88mph** [comment]
Yes thank you.  It was a lack of understanding on my part.

Dustin


Dustin N. Jenkins 
Tel/Tél: 250.363.3101 | facsimile/télécopieur: (250) 363-0045 | Dustin.Jenkins@nrc-cnrc.gc.ca 
National Research Council Canada | 5071 West Saanich Rd, Victoria BC. V9E 2E7 
Conseil national de recherches Canada | 5071, ch. West Saanich, Victoria (C.-B) V9E 2E7 
Government of Canada | Gouvernement du Canada
